experimental/dds/sequence-deprecated/src/sparsematrix.ts

```diff
@@ -73,15 +73,8 @@ export class PaddingSegment extends BaseSegment {
     }
 
     public append(segment: ISegment) {
-        if (!PaddingSegment.is(segment)) {
-            throw new Error("can only append padding segment");
-        }
-
-        // Note: Must call 'appendLocalRefs' before modifying this segment's length as
-        //       'this.cachedLength' is used to adjust the offsets of the local refs.
-        LocalReferenceCollection.append(this, segment);
-
-        this.cachedLength += segment.cachedLength;
+        assert(PaddingSegment.is(segment), "can only append padding segment");
```

Comment:
all of these methods need to call LocalReferenceCollection logic and update cachedLength, it seemed to make more sense to me to do this in the base class (once integrated, we'll need to update attribution collection info in the base class as well). I also think asserting rather than throwing errors here is reasonable, since these aren't really APIs we expect consumers to call. should be a small bundle size win since the string will be turned to an error code.

Reply:
its not necessarily a bad change, but also feel unrelated. my guess in a follow up you will also leverage this to manage attribution info, but the connection isn't clear at this point

Reply:
yeah, that's right. AttributionCollection will need similar updates to LocalReferenceCollection on segments, and current scattering of LocalReferenceCollection management to implementers of BaseSegment when they all need to do the same thing didn't seem ideal.